### PR TITLE
Project: Members Only

### PR DESCRIPTION
**Pull request by idgm5**
In this project, we build an exclusive clubhouse where members can write embarrassing posts about non-members. Inside the clubhouse, members can see who the author of a post is but, outside, they can only see the story and wonder who wrote it.

- Basic Setup
- Authentication and Posts

### CODE REVIEW

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Nice work on this project, but the following changes need to be done before moving on ⬇️** 

* [x] User can sign in ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
* [x] As a signed-in user you can create new post ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_post_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
* [x] As a signed-in user you can see post's author ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_see_author_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
* [x] User can sign_out ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_out_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
* [x] As a guest you can see only post's text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

- There are no links in the views, please add a navbar with all the links so the user of the project can navigate through it properly.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Comment by idgm5** [comment]
A user can sign_in and sin_up directly from the screen. Also, as a guest, you can only access to title and body of the post.

image

Once they signed_in with a new account or current one, they can now see the author's name of the message, in this case, is 'Isaac' the author.

image

**Comment by idgm5** [comment]
We now have a better interface with a navbar, author column (if user signed_in), and options column.

image

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Rails.application.routes.draw do
+  devise_for :users, :controllers => { registrations: 'registrations' }
+  resources :posts
```
Comment (maelfosso) [inline config/routes.rb]: Please, [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) is the indications according to the routes

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, only: %i[edit update destroy]
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb]: Not only. According to [that](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png), you've forgotten `new` and `create`

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    @posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    if user_signed_in?
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb]: Move it at `before_action` as suggested up there

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello, 
It's almost OK. Just a few changes to do.
Please, read the comments left below.
Kindly make the suggested changes and submit a new code review request.
Thanks

Mael FOSSO

File: config/routes.rb (hunk 1 above)
Comment (rubenpazch) [reply config/routes.rb]: we fix the issues

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (rubenpazch) [reply app/controllers/posts_controller.rb]: we fix the issues

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (rubenpazch) [reply app/controllers/posts_controller.rb]: we fix the issues

**Review by nikasulo — APPROVED** [review]
# Great work! @idgm5 and @rubenpazch :rocket: 
Please merge and keep moving forward, I wish you the very best and see you at the top! :medal_military: 

##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

**Pull request merged**